### PR TITLE
Treat eslint crashes as errors

### LINKED ISSUE #50: Silent failure when .eslintrc is invalid
This has caught me out a few times lately, ideally it should be treated similarly to when the JS file itself has a syntax error.

I'll try and look at adding a PR

### PR DESCRIPTION
**Pull request by glenjamin**
Fixes #50

### CODE REVIEW

**Comment by roadhump** [comment]
I tried your PR and it seems that it does not work, match on line 50 returns None. I have this message if `.eslintrc` is incorrect

```
Cannot read config file: /Users/roadhump/Projects/tx/.eslintrc
Error: JS-YAML: unexpected end of the stream within a double quoted scalar at line 6, column 1:
    
    ^
undefined
```

Can you check it, please?

**Comment by glenjamin** [comment]
Ah, a slightly different error output.

I'll expand to this case as well.

**Comment by glenjamin** [comment]
Caught that one and another as well now.

Current version handles:
* unrecognised environment
* unrecognised plugin
* invalid yaml

File: linter.py [hunk 1]
```diff
@@ -30,11 +31,39 @@ class ESLint(NodeLinter):
...
+        """
+
+        match = self.config_fail_regex.match(output)
+        if match:
+            msg = "%s in %s" % match.group(2, 1)
+            return [(match, 0, None, "Error", "", msg, None)]
+
+        match = self.crash_regex.match(output)
+        if match:
+            msg = "Eslint crashed: %s" % match.group(1)
```
Comment (roadhump) [inline linter.py]: Rename to Eslint to ESLint as an official name

File: linter.py [hunk 2]
```diff
@@ -30,11 +31,39 @@ class ESLint(NodeLinter):
...
+    def find_errors(self, output):
+        """
+        Parses errors from linter's output
+
+        We override this method to handle parsing eslint crashes
+        """
+
+        match = self.config_fail_regex.match(output)
+        if match:
+            msg = "%s in %s" % match.group(2, 1)
```
Comment (roadhump) [inline linter.py]: I suppose message `Cannot read config file: /path/.eslintrc` will be more relevant, full error message is trimmed by output and is not clear, user can run eslint in cli to get specific location of error.

**Comment by roadhump** [comment]
Also, can you squash PR commits to single, it is more convenient for history log.

**Comment by glenjamin** [comment]
Fixes made :)

**Comment by roadhump** [comment]
Nice

**Pull request merged**